src/Fantomas.CoreGlobalTool/Program.fs

```diff
@@ -69,13 +70,27 @@ let rec allFiles isRec path =
     Directory.GetFiles(path, "*.*", searchOption)
     |> Seq.filter (fun f -> isFSharpFile f && not (isInExcludedDir f))
 
+let private hasByteOrderMark file =
+    if File.Exists(file) then
+        let preamble = Encoding.UTF8.GetPreamble()
```

Comment:
This will do the wrong thing for e.g. UTF-32 encoded files. The best way I know of to discover the encoding of a file is https://stackoverflow.com/a/30393739/1462221 ; and if Fantomas wants to respect BOM then it'll have to carry around the *encoding*, rather than merely a flag noting whether a BOM is present.